### DISCUSSION TITLE
Memory do not reduce when using IndexType.IVF_SQ8

### DISCUSSION THREAD

**Discussion by markson14** [milvus-io/milvus#5176] [Q&A and General discussion] [answered]
> ENV
> total: 25M vector
> dimension: 256
> IndexType: IVF_SQ8

The overall size is 25GB in disk. It suppose to be around 8GB when it has been loaded to memory. However, I use "htop" to check the memory usage and find out it still take 25GB in memory. Can someone help me about it? 

![image](https://user-images.githubusercontent.com/30102000/117782142-ec472200-b273-11eb-9550-5e830c4fbad1.png)
`
(Status(code=0, message='Success'), {'partitions': [{'row_count': 25000000, 'segments': [...], 'tag': '_default'}], 'row_count': 25000000})
(Status(code=0, message='Describe collection successfully!'), CollectionSchema(collection_name='remoteTestCPU', dimension=256, index_file_size=256, metric_type=<MetricType: IP>))
(Status(code=0, message='Successfully'), IndexParam(_collection_name='remoteTestCPU', _index_type=<IndexType: IVF_SQ8>, _params={'nlist': 4096}))`

**Comment by yhmo** [comment, marked as answer]
When index not finished, you perform a search request, milvus will load the raw data to do brute-search since there is no index. So, if the index not finished, and you perform search request, milvus will occupy 25GB to load raw data. After all index finished, you perform search request, the milvus will load index (8GB) into memory, but since milvus cache strategy is LRU, it evict some raw data (8GB) from cache and load the index data into cache, and some other raw data(17GB) will keep in cache.
